Dockerfile

```
@@ -12,8 +12,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-ARG BASEIMAGE
-FROM ${BASEIMAGE}
+# The debian-base:v1.0.0 image built from kubernetes repository is based on
+# Debian Stretch. It includes systemd 232 with support for both +XZ and +LZ4
+# compression. +LZ4 is needed on some os distros such as COS.
+FROM k8s.gcr.io/debian-base-${TARGETARCH}:v2.0.0
+
+ARG TARGETOS
+ARG TARGETARCH
```

```suggestion
ARG TARGETARCH
FROM k8s.gcr.io/debian-base-${TARGETARCH}:v2.0.0

ARG TARGETOS
```

AFAIK this placement [is required](https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact)?

---

ah, sorry, missed the fact that TARGETOS is predefined by BuildKit :facepalm: 

Maybe just to clarify, for anyone - like me - trying to build with other builders like buildah?
```suggestion
FROM k8s.gcr.io/debian-base-${TARGETARCH}:v2.0.0

# pull BuildKit global vars into the build stage
ARG TARGETOS
ARG TARGETARCH
```